### PR TITLE
Audio: DCblock: Tune: Move dcblock setup scripts to module directory

### PR DESCRIPTION
**Pull request by singalsu**

### CODE REVIEW

**Comment by cujomalainey** [comment]
If this is part of a larger migration of code can you link a top-level bug?

**Review by kv2019i — COMMENTED** [review]
@singalsu I know why you are doing this, but I would not figure this out from the commit messages. Please add a brief comment why this move is made, especially as this is first of its kind.

**Comment by singalsu** [comment]
> If this is part of a larger migration of code can you link a top-level bug?

Yes, I'll add the reason to commit message. The purpose is to collect all module rated into same directory for easier access. Same has been already done for kconfig and IPC .toml. The functionality is not changed, though I added some checks for file opens since moving the files with triggered some fopen() errors with old paths.

File: tools/tune/common/tplg_write.m [hunk 1]
```diff
@@ -15,6 +15,11 @@ function tplg_write(fn, blob8, name, comment, howto)
 
 %% Write blob
 fh = fopen(fn, 'w');
+if fh < 0
```
Comment (singalsu) [inline tools/tune/common/tplg_write.m:18]: Changes to common could be in another PR maybe? I realized I had in another PR similar error adds to other exports.

For now I'm not moving tools/tune/common -- I don't know where they could go, or is there any benefit to move the helper functions closer to src/audio modules.

File: src/audio/dcblock/tune/example_dcblock.m [hunk 2]
```diff
@@ -31,13 +31,16 @@ function example_dcblock()
...
+sof_tplg = fullfile(sof_tools, 'topology');
+sof_ctl = fullfile(sof_tools, 'ctl');
+tplg1_fn = sprintf("%s/topology1/m4/dcblock_coef_%s.m4", sof_tplg, prm.id); % Control Bytes File
+tplg2_fn = sprintf("%s/topology2/include/components/dcblock/%s.conf", sof_tplg, prm.id);
 % Use those files with sof-ctl to update the component's configuration
-blob3_fn = sprintf("../../ctl/ipc3/dcblock/coef_%s.blob", prm.id); % Blob binary file
-alsa3_fn = sprintf("../../ctl/ipc3/dcblock/coef_%s.txt", prm.id); % ALSA CSV format file
-blob4_fn = sprintf("../../ctl/ipc4/dcblock/coef_%s.blob", prm.id); % Blob binary file
-alsa4_fn = sprintf("../../ctl/ipc4/dcblock/coef_%s.txt", prm.id); % ALSA CSV format file
+blob3_fn = sprintf("%s/ipc3/dcblock/coef_%s.blob", sof_ctl, prm.id); % Blob binary file
```
Comment (cujomalainey) [inline src/audio/dcblock/tune/example_dcblock.m]: `prm.id`? what is `prm` and what is `id`

File: src/audio/dcblock/tune/sof_dcblock_build_blob.m [hunk 3]
```diff
@@ -1,4 +1,4 @@
-function blob8 = dcblock_build_blob(R_coeffs, endian, ipc_ver)
+function blob8 = sof_dcblock_build_blob(R_coeffs, endian, ipc_ver)
```
Comment (cujomalainey) [inline src/audio/dcblock/tune/sof_dcblock_build_blob.m:1]: why does the prefix matter if we are already isolated to our own module?

**Review by cujomalainey — CHANGES_REQUESTED** [review]

File: src/audio/dcblock/tune/sof_dcblock_build_blob.m (hunk 3 above)
Comment (singalsu) [reply src/audio/dcblock/tune/sof_dcblock_build_blob.m:1]: Good question, any thoughts @lrgirdwo, should we leave the original file names?

File: src/audio/dcblock/tune/example_dcblock.m [hunk 4]
```diff
@@ -49,7 +49,7 @@ function dcblock_blob_calculate(prm)
 	R_coeffs = R * ones(1, channels);
 end
 
-addpath ./../common
+dcblock_paths(1);
```
[the remark this answers is not in the document]
Comment (singalsu) [reply src/audio/dcblock/tune/example_dcblock.m]: Yep, `false` and `true` are available.

File: src/audio/dcblock/tune/example_dcblock.m (hunk 2 above)
Comment (singalsu) [reply src/audio/dcblock/tune/example_dcblock.m]: Prm was earlier added to collect the parameters into an easy to pass struct. Id is a string to append to blob file name dcblock/coef_<id>.conf. Current usages are "default" and  like "50hz_48kHz". Currently there are blobs for 20, 30, 40, 50, 80, 100, 200 Hz cut-off @ 48 kHz.

File: src/audio/dcblock/tune/example_dcblock.m (hunk 4 above)
[the remark this answers is not in the document]
Comment (cujomalainey) [reply src/audio/dcblock/tune/example_dcblock.m]: please use them then, makes it a bit clearer

**Review by lgirdwood — COMMENTED** [review]
@singalsu I guess we need to set the PATH for matlab to find files/functions that are not in the current dir ? Btw, does Matlab/Octave has a default set of directories it uses for functions ?

File: src/audio/dcblock/tune/example_dcblock.m (hunk 4 above)
[the remark this answers is not in the document]
Comment (singalsu) [reply src/audio/dcblock/tune/example_dcblock.m]: Yep, I will update this, now busy with another topic but soon!

**Comment by singalsu** [comment]
> @singalsu I guess we need to set the PATH for matlab to find files/functions that are not in the current dir ? Btw, does Matlab/Octave has a default set of directories it uses for functions ?

Yep, there's functions addpath and rmpath for it. The default path is just current directory and Matlab/Octave internal toolboxes paths. The user can also alter paths via .matlabrc and .octaverc.

**Review by cujomalainey — APPROVED** [review]
